src/common.py

```diff
@@ -251,6 +251,68 @@ def v_space(n: int, col=None) -> None:
         else:
             st.write("#")
 
+def display_large_dataframe(df, 
+                            chunk_sizes: list[int] = [100, 1_000, 10_000]):
+    """
+    Displays a large DataFrame in chunks with pagination controls and row selection.
+
+    Args:
+        df: The DataFrame to display.
+        chunk_sizes: A list of chunk sizes to choose from.
+
+    Returns:
+        Selected rows from the current chunk.
+    """
+    # Dropdown for selecting chunk size
+    chunk_size = st.selectbox("Select Number of Rows to Display", chunk_sizes)
```

Comment:
I would add an `on_change` callback such that `current_chunk` is reset if a new selection is made. I tested it and went to row 700 using `chunk_size=100`. When changing to `chunk_size=1000` I was at row 7000 instead of the previous 700.

Reply:
Ohh good catch! Done.